clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -371,6 +375,38 @@ public synchronized AbstractStorageEngine openStore(
     return engine;
   }
 
+  public synchronized void checkWhetherStoragePartitionsShouldBeKeptOrNot(SafeHelixManager manager) {
+    if (getStorageEngineRepository() == null || manager == null) {
+      return;
+    }
+    for (AbstractStorageEngine storageEngine: getStorageEngineRepository().getAllLocalStorageEngines()) {
+      String storageEngineName = storageEngine.getStoreVersionName();
+      String storeName = Version.parseStoreFromKafkaTopicName(storageEngineName);
+      PropertyKey.Builder propertyKeyBuilder =
+          new PropertyKey.Builder(configLoader.getVeniceClusterConfig().getClusterName());
+      SafeHelixDataAccessor helixDataAccessor = manager.getHelixDataAccessor();
+      IdealState idealState = helixDataAccessor.getProperty(propertyKeyBuilder.idealStates(storeName));
+
+      if (idealState == null) {
+        return;
```

Comment:
This seems wrong.  This will abort the whole function, but we would want to iterate on the remainder of the storage engines to determine if we should keep them.  I think instead of return, we should use continue